### PR TITLE
CI

### PR DESCRIPTION
**Pull request by ecoologic**
* Refresh language and libraries to use PHP 7.4
* Fixed some tests failing due to now using PHP 7.4
* Fixed some tests failing due to being broken in the first place
* Skipped some tests to get a green build
* Added CI running tests
* CI fails when it should, see [here](https://github.com/zendesk/zendesk_api_client_php/actions/runs/6426888352/job/17451723058?pr=508)
* Ignoring live specs for now

This is a decent milestone and a good time to split the work. Usually I don't leave `TODOs` in PRs but I will follow up with another PR to solve those issues.

```
OK, but incomplete, skipped, or risky tests!
Tests: 261, Assertions: 2284, Skipped: 6.
```

There's more than 6 tests to fix, the tough ones, but less than a dozen.

### CODE REVIEW

File: tests/bootstrap.php [hunk 1]
```diff
@@ -1,6 +1,9 @@
 <?php
 
+// TODO: Don't fail using deprecated code
+// error_reporting(E_ALL ^ E_DEPRECATED);
 error_reporting(- 1);
```
Comment (ecoologic) [inline tests/bootstrap.php:5]: I will allow the use of deprecated code, at least as a milestone, but if I do that now the tests keep going and breaks the whole test execution, so I'll do that separately.

**Review by fbvilela — APPROVED** [review]
LGTM! 👍

File: tests/Zendesk/API/UnitTests/Middleware/RetryHandlerTest.php [hunk 2]
```diff
@@ -36,7 +36,7 @@ public function testExceptionsRetry($exception, $success)
             }
         ];
         $client = $this->mockApiResponses([
-            new $exception('', new Request('GET', '')),
+            new $exception('', new Request('GET', ''), new Response()),
```
Comment (yoshdog) [inline tests/Zendesk/API/UnitTests/Middleware/RetryHandlerTest.php:39]: 👍

**Review by yoshdog — APPROVED** [review]

**Comment by cryptomail** [comment]
🥳

**Pull request merged**